scanner/decisions/0010-separate-versioning-for-vuln-bundles.md

```diff
@@ -0,0 +1,175 @@
+3.  A StackRox release, `release/<stackrox-release>`, example: `release/4.6.0`
+
+Both (1.) and (2.) will determine a specific commit to use to build the vulnerability exporter.
+
+(3.) will select a desired release build based on a StackRox target release.  The workflow will find the best matching Git tag for the specified `<stackrox-release>`, by checking if the version exists as a tag or matching it against release candidate (`-rc`) or patch (`.x`) tags.
+
+For each version in the mapping, a bundle will be built based on the determined git ref and stored at `https://definitions.stackrox.io/v4/<bundle-version>/vulnerabilities.zip`.
+
+In every release cut, the key specified in `scanner/VULNERABILITY_VERSION` will be updated in the mapping to the current release reference (e.g., `release/4.6.0`).  Notice that the key is still updated if the version is the same.   If it's new, it will be added, and the previous version will stay.
+
+For example, in release cut 4.6 we would look into `scanner/VULNERABILITY_VERSION`, let's say its contents are:
+
+```
+$ cat canner/VULNERABILITY_VERSION
+v1
+```
+
+The mapping should be updated so `v1` points to the current release branch reference.  Let's say the mapping was:
+
+bundle version | git reference
+--- | ---
+`dev` | `master`
+`v1` | `release/4.5.0`
+
+Then, `v1` is updated to the new release (in this example, `4.6.0`):
+
+bundle version | git reference
+--- | ---
+`dev` | `master`
+`v1` | `release/4.6.0`
+
+If in 4.6 the bundle version was updated to:
+
+```
+$ cat canner/VULNERABILITY_VERSION
+v2
+```
+
+On 4.6.0 release cut, the final mapping would be:
+
+bundle version | git reference
+--- | ---
+`dev` | `master`
+`v1` | `release/4.5.0`
+`v2` | `release/4.6.0`
+
+#### Offline mode
+
+Per-release offline bundles will continue to be published.  They will be versioned by StackRox releases, but they will not be based on Y-stream (`4.5`) but on Z-stream patch releases (`4.5.0`).  The bundle will contain the correct vulnerability bundle for that particular Z-stream release.  The `manifest.json` file will continue to have a "version" attribute pointing out the Y-stream for that offline bundle, but two other fields will be added: "vulnerability_version" and "release_versions".  For example, for patch release `4.6.1` the offline bundle could be:
+
+```
+{
+  "version": "4.6",
+  "created": "2024-08-20T21:00:30+00:00",
+  "release_versions": ["4.6.0", "4.6.1", ...],
+  "vulnerability_version": "v1"
+}
+```
+
+The central definitions handler will validate the offline bundle POST against `release_versions` to ensure the running Central version is supported.  And, when consuming the offline bundle, against "vulnerability_version", guaranteeing the bundle supports the Scanner requested schema version.
+
+We will continue to offer the latest offline bundles.  They will contain all offline bundles versioned by schema, and the existing `4.5` bundle to continue to be added for backward compatibility.  Example:
```

Comment:
> We will continue to offer the latest offline bundles.

This refers to install.stackrox.io/... right?

Reply:
Yes, not the bundle in `definitions.stackrox.io`, named `scanner-v4-defs-*.zip`.